### PR TITLE
[8.0] File Attachments do not show in interface

### PR DESCRIPTION
**Pull request by ghost**
Description of the issue/feature this PR addresses:
An attachment has been saved to a shipment form but is not available after the upgrade to odoo V8

Current behavior before PR:
attachment is not available in v8 in a shipment form

Desired behavior after PR is merged:
Attachment is available

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

To find the correct attachment files the id and table name are searched. This causes missing attachment becaut the tables stock.picking.in and stock.picking.out are merged into stock.picking

### CODE REVIEW

**Comment by hbrunn** [comment]
[edit] stupid question.

looks good!

File: openerp/addons/base/migrations/8.0.1.3/pre-migration.py [hunk 1]
```diff
@@ -47,3 +47,17 @@ def migrate(cr, version):
         cr, 'ir_ui_view', 'type', [
             'calendar', 'diagram', 'form', 'gantt', 'graph', 'kanban',
             'qweb', 'search', 'tree'])
+    
+    #The tables stock.picking.in and stock.picking.out are merged into stock.picking
+    openupgrade.logged_query(
+        cr, """
+        UPDATE ir_attachment
+        SET res_model = 'stock.picking'
+        WHERE res_model = 'stock.picking.out';
```
Comment (hbrunn) [inline openerp/addons/base/migrations/8.0.1.3/pre-migration.py]: make this one query

File: openerp/addons/base/migrations/8.0.1.3/pre-migration.py [hunk 2]
```diff
@@ -47,3 +47,12 @@ def migrate(cr, version):
         cr, 'ir_ui_view', 'type', [
             'calendar', 'diagram', 'form', 'gantt', 'graph', 'kanban',
             'qweb', 'search', 'tree'])
+    
+    #The tables stock.picking.in and stock.picking.out are merged into stock.picking
```
Comment (pedrobaeza) [inline openerp/addons/base/migrations/8.0.1.3/pre-migration.py]: This comment should be have a space between the `#` and the first word and also check 80 cols limit.

File: openerp/addons/base/migrations/8.0.1.3/pre-migration.py [hunk 3]
```diff
@@ -47,3 +47,12 @@ def migrate(cr, version):
...
+    openupgrade.logged_query(
+        cr, """
+        UPDATE ir_attachment
+        SET res_model = 'stock.picking'
+        WHERE res_model = 'stock.picking.in' OR
```
Comment (pedrobaeza) [inline openerp/addons/base/migrations/8.0.1.3/pre-migration.py]: Better to use:
```
WHERE res_model in ('stock.picking.in', 'stock.picking.out')
```

**Comment by pedrobaeza** [comment]
@hbrunn @StefanRijnhart, don't you feel the need for an `update_model_names` method in openupgradelib that handles these things?:

* Attachments
* Reference fields entries
* Other references through base objects

**Review by pedrobaeza — APPROVED** [review]
Squash on merge

**Comment by hbrunn** [comment]
this need was felt a long time ago: https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L362

it's just that this specific migration script doesn't seem to use it

**Comment by pedrobaeza** [comment]
Wow, but that something that we have to use in all places!! We don't know when we are going to have a reference field pointing to our model, so this one should be the default renaming method.

**Comment by pedrobaeza** [comment]
Anyway, I'm seeing that the reference fields are not covered by this method, and also ir.ui.view and other base models that uses model name as a text column.

**Comment by hbrunn** [comment]
reading the code, we don't rename anything in the stock migration. @M2tec better use aforementioned function and apply it  to both models in https://github.com/M2tec/OpenUpgrade/blob/50c02fb3cfed1cdca29e5ecf9bcd0690f21f8d42/addons/stock/migrations/8.0.1.1/pre-migration.py#L178

**Comment by pedrobaeza** [comment]
The problem this time is going to be with duplicated field entries, as there are fields with the same name in the three tables.

**Comment by ghost** [comment]
Ok I will look into the function when I have time

**Comment by ghost** [comment]
If you insert this in stock pre-migrations script:
openupgrade.rename_models(cr, [('stock.picking.in','stock.picking')])

you get the following error:

IntegrityError: duplicate key value violates unique constraint "ir_model_obj_name_uniq"
DETAIL:  Key (model)=(stock.picking) already exists.

**Review by hbrunn — APPROVED** [review]
yes, that's what @pedrobaeza wrote before. Leave it as it is

**Review by StefanRijnhart — APPROVED** [review]
Thank you!

**Comment by ghost** [comment]
Ok cool, first patch in opensource software :). Thanks guys

**Comment by pedrobaeza** [comment]
I hope this is the first of many to come :wink:

**Pull request merged**